src/main/java/com/avispl/symphony/dal/infrastructure/management/qsc/qsyscore/QSYSCoreAggregatorCommunicator.java

```diff
@@ -495,37 +492,182 @@ public void controlProperty(ControllableProperty controllableProperty) throws Ex
 			if (qrcCommunicator == null) {
 				initQRCCommunicator();
 			}
-
 			String property = controllableProperty.getProperty();
 			String value = String.valueOf(controllableProperty.getValue());
-
+			String deviceId = controllableProperty.getDeviceId();
 			if (this.logger.isDebugEnabled()) {
 				this.logger.debug("controlProperty property " + property);
 				this.logger.debug("controlProperty value " + value);
 			}
 
-			String[] splitProperty = property.split(QSYSCoreConstant.HASH);
-
-			// Ex: Gain:Named Component#Gain Value Control
-			// metricName = Gain Value Control
-			// namedComponent = Named Component
-			String metricName = splitProperty[1];
-			List<String> splitComponent = Arrays.asList(splitProperty[0].split(QSYSCoreConstant.COLON, 2));
-			switch (splitComponent.get(0)) {
-				case QSYSCoreConstant.GAIN:
-					String gainComponent = splitComponent.size() > 1 ? splitComponent.get(1) : splitComponent.get(0);
-					gainControl(metricName, gainComponent, value);
-					updateGainControlByMetricName(metricName, value, property);
-					break;
-				default:
-					logger.debug("Component Name doesn't support: " + metricName);
+			if (StringUtils.isNotNullOrEmpty(deviceId)) {
+				mapOfIdAndAggregatedDeviceList.entrySet().stream().filter(device -> device.getKey().equalsIgnoreCase(deviceId)).filter(device -> "Plugin".equalsIgnoreCase(device.getValue().getType()))
+						.findFirst().ifPresent(device -> {
+							if (deviceId.contains("Sennheiser")) {
+								SennheiserDeviceMetric sennheiserDevice = EnumTypeHandler.getPropertiesByName(SennheiserDeviceMetric.class, property);
+								switch (sennheiserDevice) {
+									case AUDIO_LEVEL:
+										populateSennheiserControl(deviceId, property, value);
+										device.getValue().getStats().put(QSYSCoreConstant.AUDIO_LEVEL_CURRENT_VALUE, value);
+										break;
+									case LED_BRIGHTNESS:
+										populateSennheiserControl(deviceId, property, value);
+										device.getValue().getStats().put(QSYSCoreConstant.LED_BRIGHTNESS_CURRENT_VALUE, value);
+										break;
+									case MUTE_COLOR:
+									case ON_COLOR:
+									case NOISE_LEVEL:
+									case SOUND_PROFILE:
+										String result = "\"" + value + "\"";
+										populateSennheiserControl(deviceId, property, result);
+										break;
+									default:
+										populateSennheiserControl(deviceId, property, value);
+										break;
+								}
+							} else if (deviceId.contains("MiddleAtlantic")) {
+								MiddleAtlanticMetric middleAtlantic = EnumTypeHandler.getPropertiesByName(MiddleAtlanticMetric.class, property);
+								switch (middleAtlantic) {
+									case OUTLET_1_NAME:
```

Comment:
is it ok that none of these cases are having break; statement? 
I assume we won't expect these control commands here yet, but Im worried that each of these will fall down to the restart command